### ISSUE TITLE
Allow duplicate cluster urls.

### ISSUE THREAD

**Issue by servo1x** [argoproj/argo-cd#2288]
**Is your feature request related to a problem? Please describe.**
Feature. We are deploying applications to clusters managed by another team - and they only provide service accounts locked to a specific namespace. So we have multiple cluster configs going to the same cluster, using different service accounts for different namespaces.

**Describe the solution you'd like**
Change the unique id to the cluster name instead of the cluster url.

**Workaround.**

We append a query parameter to the cluster url, ie. `?__uid=unique-id`.

Related: https://github.com/argoproj/argo-cd/issues/1548

**Comment by alexec** [comment]
Interesting. @jessesuen In other tickets (#1814 ?) people ask for cluster alias, so doing both would solve this. Large piece of work.

**Comment by acherla** [comment]
I dont think this actually solves the issue.  The user is explicitly requesting the ability to add the same cluster api url with different alias'.  Right now since ArgoCD keys off of the api server url, it results in the controller picking up the incorrect cluster.  

The user is likely asking for this feature because they want the ability to add multiple alias' for the same cluster due to namespaced restrictions.

I have a similar requirement as well and noticed this same issue since argocd keys off of the cluster api server url.

**Comment by andrii-korotkov-verkada** [comment]
Closed since there's been quite a long time since updates and it's unclear if this is still relevant. Feel free to re-open/re-submit the issue if still relevant.
